commit 5c316c1ea1c1fc0b67b82eb39eeca9cd2ca9d33c
Author: Alexey Kardashevskiy <aik@ozlabs.ru>
Date:   Fri Mar 17 00:48:26 2017 +0000

    vfio/spapr: Postpone allocation of userspace version of TCE table
    
    [ Upstream commit 39701e56f5f16ea0cf8fc9e8472e645f8de91d23 ]
    
    The iommu_table struct manages a hardware TCE table and a vmalloc'd
    table with corresponding userspace addresses. Both are allocated when
    the default DMA window is created and this happens when the very first
    group is attached to a container.
    
    As we are going to allow the userspace to configure container in one
    memory context and pas container fd to another, we have to postpones
    such allocations till a container fd is passed to the destination
    user process so we would account locked memory limit against the actual
    container user constrainsts.
    
    This postpones the it_userspace array allocation till it is used first
    time for mapping. The unmapping patch already checks if the array is
    allocated.
    
    Signed-off-by: Alexey Kardashevskiy <aik@ozlabs.ru>
    Reviewed-by: David Gibson <david@gibson.dropbear.id.au>
    Acked-by: Alex Williamson <alex.williamson@redhat.com>
    Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>
    Signed-off-by: Sasha Levin <alexander.levin@verizon.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/drivers/vfio/vfio_iommu_spapr_tce.c
===================================================================
--- linuxtest.orig/drivers/vfio/vfio_iommu_spapr_tce.c
+++ linuxtest/drivers/vfio/vfio_iommu_spapr_tce.c
@@ -543,6 +543,12 @@ static long tce_iommu_build(struct tce_c
 	unsigned long hpa;
 	enum dma_data_direction dirtmp;
 
+	if (!tbl->it_userspace) {
+		ret = tce_iommu_userspace_view_alloc(tbl);
+		if (ret)
+			return ret;
+	}
+
 	for (i = 0; i < pages; ++i) {
 		unsigned long offset = tce & IOMMU_PAGE_MASK(tbl) & ~PAGE_MASK;
 
Index: linuxtest/dummy/rpi_1433_5c316c1ea1c1fc0b67b82eb39eeca9cd2ca9d33c.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1433_5c316c1ea1c1fc0b67b82eb39eeca9cd2ca9d33c.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
